frontend/src/components/MapView/Layers/CompositeLayer/index.tsx

```diff
@@ -26,7 +59,34 @@ const CompositeLayer = ({ layer }: Props) => {
     }
   }, [dispatch, layer, data, date]);
 
-  return <>test</>;
+  const { adminBoundaryLimitPolygon } = useLayers();
+
+  if (data && adminBoundaryLimitPolygon) {
+    const filteredData = {
```

Comment:
Another option is to use mapbox filtering directling using the `within` command. When I used it locally it seemed slower but it could be worth checkin on deployed versions. Here is the code for reference:

```
  const { adminBoundaryLimitPolygon } = useLayers();

  const paintProps: (
    opacity: number | undefined,
  ) => CircleLayerSpecification['paint'] = (opacity?: number) => ({
    'circle-opacity': [
      'case',
      ['within', adminBoundaryLimitPolygon],
      opacity || 0.5,
      0,
    ],
    'circle-color': [
      'interpolate',
      ['linear'],
      ['get', 'value'],
      -2,
      '#FFFF33',
      2,
      '#FF8433',
    ],
    'circle-radius': [
      'interpolate',
      ['exponential', 2],
      ['zoom'],
      0,
      1 * scale,
      15,
      16 * scale,
    ],
  });
```

Reply:
Note that it also seems to accept only one feature / polygon at a time, so we still need to do the merge on our own...